api/server.py

```diff
@@ -227,1092 +240,197 @@ def get_static_files(path):
-            0,
-            {
-                "$match": {
-                    'module_name': module_name
-                }
-            }
-        )
-    try:
-        return jsonify(
-            aggregate_function(connector.credential_events, module_query)
-        ), 200
+        return json.loads(json_util.dumps(files_list)), 200
     except Exception:
-        return flask_null_array_response()
+        abort(500)
 
 
-@app.route("/api/events/most-usernames-used", methods=["GET"])
-def top_usernames_used():
+@app.route("/api/pcap/download", methods=["GET"])
+def download_file():
     """
-    Get top usernames used according to module
-
-    Returns:
-        JSON/Dict of top usernames used
+    Download PCAP files
     """
-    module_name = get_value_from_request("module_name")
-    module_query = [
-        group_by_ip_dest_and_username,
-        {
-            "$skip": fix_skip(
-                get_value_from_request("skip")
-            )
-        },
-        {
-            "$limit": fix_limit(
-                get_value_from_request("limit")
-            )
-        }
-    ]
-    if module_name:
-        module_query.insert(
-            0,
-            {
-                "$match": {
-                    'module_name': module_name
-                }
-            }
-        )
     try:
-        return jsonify(
-            aggregate_function(connector.credential_events, module_query)
-        ), 200
-    except Exception:
-        return flask_null_array_response()
-
-
-@app.route("/api/events/most-passwords-used", methods=["GET"])
-def top_passwords_used():
-    """
-    Get top passwords used according to module
+        file_id = ObjectId(get_value_from_request("_id"))
+        fs = connector.ohp_file_archive_gridfs.get(file_id)
 
-    Returns:
-        JSON/Dict of top passwords used
-    """
-    module_name = get_value_from_request("module_name")
-    module_query = [
-        group_by_ip_dest_and_password,
-        {
-            "$skip": fix_skip(
-                get_value_from_request("skip")
-            )
-        },
-        {
-            "$limit": fix_limit(
-                get_value_from_request("limit")
-            )
-        }
-    ]
-    if module_name:
-        module_query.insert(
-            0,
-            {
-                "$match": {
-                    'module_name': module_name
-                }
-            }
-        )
-    try:
-        return jsonify(
-            aggregate_function(connector.credential_events, module_query)
+        return send_file(
+            fs,
+            attachment_filename=fs.filename,
+            as_attachment=True,
+            mimetype=fs.content_type
         ), 200
+
     except Exception:
-        return flask_null_array_response()
+        return abort(404)
 
 
-@app.route("/api/events/module-names", methods=["GET"])
+@app.route("/api/core/list/modules", methods=["GET"])
 def all_module_names():
     """
     Get top passwords used according to module
```

Comment:
doc string of this also needs to be changed

Reply:
ok